allow expanded NPC and sprite tables

tolmar opened the pull request:
This goes along with a change I made in the CoilSnake repository. NPCs and sprites can now be expanded.

EbProjectEditor really needs a cleanup, and this change is not it. It's just what's needed to get the feature working.

This was also kind of complex (doesn't look like it but EbProjectEditor is silently eating most stack traces somewhere), so please double check the code and test it to make sure I pulled all the files, etc.

Also I'm not sure if SpriteLoader is ever used? I left my changes in because they're probably about right. It's hard to tell what's going on because something keeps silently eating stack traces.

tolmar committed:
allow expanded NPC and sprite tables

charasyn commented on src/ebhack/MapEditor.java:
```diff
@@ -2144,19 +2142,23 @@ private void importNPCs(File f) {
-									"eb.SpriteGroupModule",
-									"SpriteGroups/"
-											+ ToolModule.addZeros(i + "", 3))));
+					// TODO: this was using proj.getFilename. Why does that even exist?
+					String filename = proj.getDirectory() + File.separator + "SpriteGroups" + File.separator
+							+ ToolModule.addZeros(i + "", 3) + ".png";
+					File file = new File(filename);
+					if (!file.exists()) {
+						break;
+					}
```
I have an idea for improving this.. I'll take a crack at it. Just installing IDEA now

charasyn committed:
Update `getFilename` to better handle missing files

Now, if we try to open a SpriteGroup file that's not in the project, and one exists following the default naming convention, it will automatically be added to the project.
This also changes MapEditor.java to use the new getFilenameOrNull to load SpriteGroups.

tolmar committed:
Merge pull request #1 from charasyn/expansions

Update `getFilename` to better handle missing files

tolmar commented:
Brought in your change, it's a good one.

charasyn reviewed (approved):
LGTM. I've tested locally and it seems to work as expected. Let's merge it!
